src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -200,9 +200,18 @@ void PollingResolver::GetResultStatus(absl::Status status) {
         gpr_log(GPR_INFO, "[polling resolver %p] retrying immediately", this);
       }
     }
-    Ref(DEBUG_LOCATION, "next_resolution_timer").release();
-    GRPC_CLOSURE_INIT(&on_next_resolution_, OnNextResolution, this, nullptr);
-    grpc_timer_init(&next_resolution_timer_, next_try, &on_next_resolution_);
+    next_resolution_timer_handle_ =
```

Comment:
Now that this code is sizable and duplicated, it might be worth wrapping is inside a method like 'ScheduleNextResolutionTimer(timeout)`, to be used here and on line 246.